From e61238022c903aaec88609454f602e3fda0cd623 Mon Sep 17 00:00:00 2001
From: Andrey Smirnov <andrew.smirnov@gmail.com>
Date: Sat, 29 Oct 2022 15:25:21 -0700
Subject: [PATCH 18/21] usb: dwc3: Drop "usb-role-swtich" from
 dwc3_pci_mr_properties

Having both "usb-role-swtich" and "linux,extcon-name" is a conflicting
configuration since the two are mutually exclusive. This wasn't a
problem before since "linux,extcon-name" had a priority, but recent
change in the kernel changed that, breaking Deck's role detection
integration.

Fix this by dropping "usb-role-switch" (and "role-switch-default-mode"
with it), to allow dwc3 driver to use extcon as it should.

(cherry picked from commit 498d5dccfac3e71b95fe3b9d6a156f15b6028aec)
Signed-off-by: Cristian Ciocaltea <cristian.ciocaltea@collabora.com>
---
 drivers/usb/dwc3/dwc3-pci.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/usb/dwc3/dwc3-pci.c b/drivers/usb/dwc3/dwc3-pci.c
index 393a2efac0c7..21a764d30bff 100644
--- a/drivers/usb/dwc3/dwc3-pci.c
+++ b/drivers/usb/dwc3/dwc3-pci.c
@@ -176,8 +176,6 @@ static const struct property_entry dwc3_pci_amd_properties[] = {
 
 static const struct property_entry dwc3_pci_mr_properties[] = {
 	PROPERTY_ENTRY_STRING("dr_mode", "otg"),
-	PROPERTY_ENTRY_BOOL("usb-role-switch"),
-	PROPERTY_ENTRY_STRING("role-switch-default-mode", "host"),
 	PROPERTY_ENTRY_BOOL("linux,sysdev_is_parent"),
 	PROPERTY_ENTRY_STRING("linux,extcon-name", "VLV0100:00"),
 	{}
-- 
2.44.0

